components/Dialogs/tests/unit/MDCAlertControllerTests.m

```diff
@@ -109,13 +115,34 @@ - (void)tearDown {
   [super tearDown];
 }
 
-- (void)testInit {
+- (void)testCommonInitializationsFor:(MDCAlertController *)alert {
   // Then
-  XCTAssertNotNil(self.alert.actions);
-  XCTAssertNotNil(self.alert.title);
-  XCTAssertNotNil(self.alert.message);
-  XCTAssertTrue(self.alert.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable);
-  XCTAssertEqualObjects(self.alert.shadowColor, UIColor.blackColor);
+  XCTAssertNotNil(alert.actions);
+  XCTAssertNotNil(alert.title);
+  XCTAssertTrue(alert.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable);
+  XCTAssertEqualObjects(alert.shadowColor, UIColor.blackColor);
+}
+
+/**
+ Verifies that the message init does call initialize other variables correctly and configures, as per
+ the common init, and the message property
+ */
+- (void)testMessageInit {
+  // Then
+  [self testCommonInitializationsFor:self.alert];
+  XCTAssertNotNil(alert.message);
+  XCTAssertNil(alert.attributedMessage);
+}
+
+/**
+ Verifies that the attributedMessage init does call initialize other variables correctly and configures, 
+ as per the common init, and the attributedMessage property
+ */
+- (void)testAttributedMessageInit {
+  // Then
+  [self testCommonInitializationsFor:self.attributedAlert];
```

Comment:
One minor thing: for clarity of testing we prefer keeping all asserts in the test method itself.

```objc
- (void)testMessageInit {
  // Then
  XCTAssertNotNil(self.alert.actions);	  XCTAssertNotNil(alert.actions);
  XCTAssertNotNil(self.alert.title);	  XCTAssertNotNil(alert.title);
  XCTAssertNotNil(self.alert.message);	  XCTAssertTrue(alert.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable);
  XCTAssertTrue(self.alert.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable);	  XCTAssertEqualObjects(alert.shadowColor, UIColor.blackColor);
  XCTAssertNotNil(alert.message);
  XCTAssertNil(alert.attributedMessage);
}
```

etc... for the other method.

Reply:
On it